alchemist/alchemist-maps/src/main/java/at/jku/traces/json/JKUJSONLoader.java

```diff
@@ -23,15 +24,41 @@
 
 import com.google.common.base.Charsets;
 import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
 
+import it.unibo.alchemist.model.interfaces.GPSPoint;
 import it.unibo.alchemist.model.interfaces.GPSTrace;
 
 /**
  */
 public final class JKUJSONLoader implements Serializable {
 
     private static final long serialVersionUID = 7144531714361675479L;
-    private static final Gson GSON = new Gson();
+    private static final Gson GSON = new GsonBuilder()
+            .registerTypeAdapter(GPSPoint.class, new JsonDeserializer<GPSPoint>() {
+                @Override
+                public GPSPoint deserialize(final JsonElement json, final Type typeOfT, final JsonDeserializationContext context) throws JsonParseException {
+                    if (json.isJsonObject()) {
+                        final JsonObject obj = json.getAsJsonObject();
+                        try {
```

Comment:
Don't use exceptions as control flow: check that these objects are doubles before calling the getter